tests/syscall/xattr/xattr.bats

```diff
@@ -219,7 +219,41 @@ function teardown() {
 	rm -rf /mnt/scratch/test
 }
 
-@test "l*xattr high-util: nixos use-case" {
+@test "allow-trusted-xattr disabled" {
+
+	rm -rf /mnt/scratch/test
+	mkdir -p /mnt/scratch/test
+	chown 165536:165536 /mnt/scratch/test
+
+	# deploy a sys container, turn off "allow-trusted-xattr" for it
+	local syscont=$(docker_run --rm -e "SYSBOX_ALLOW_TRUSTED_XATTR=FALSE" -v /mnt/scratch/test:/mnt ${CTR_IMG_REPO}/alpine-docker-dbg:latest tail -f /dev/null)
+
+	# the attr package brings the setfattr and getfattr utils
+	docker exec "$syscont" sh -c "apk add attr"
+	[ "$status" -eq 0 ]
+
+	docker exec "$syscont" sh -c "mkdir /mnt/tdir && touch /mnt/tdir/tfile"
+	[ "$status" -eq 0 ]
+
+	# setting "trusted.overlay.opaque" xattr should fail now
+	docker exec "$syscont" sh -c 'setfattr -n trusted.overlay.opaque -v "y" /mnt/tdir/tfile'
+	[ "$status" -eq 1 ]
+	docker exec "$syscont" sh -c 'setfattr -n trusted.another -v "another value" /mnt/tdir/tfile'
+	[ "$status" -eq 1 ]
+	docker exec "$syscont" sh -c 'cd /mnt && getfattr -d -m "trusted\." tdir/tfile'
+	[ "$status" -eq 0 ]
+	[[ "$output" == "" ]]
+
+	# setting "user.*" xattr is fine
+	docker exec "$syscont" sh -c 'setfattr -n user.x -v "user value" /mnt/tdir/tfile'
```

Comment:
Won't this instruction fail in kernels < 5.11?

Reply:
No; this is not for user.overlay.opaque; it's an arbitrary user.* xattr we are choosing, so that works on older kernels too.